### PR TITLE
DS-3782 upgrade commons collections to version 4

### PR DESCRIPTION
**Pull request by Raf-atmire**
This is the suggested upgrade to commons collections version 4 as requested on jira ticket: https://jira.duraspace.org/browse/DS-3782

### CODE REVIEW

File: dspace-api/src/main/java/org/dspace/authenticate/IPAuthentication.java [hunk 1]
```diff
@@ -176,7 +177,7 @@ public boolean isImplicit()
     {
         if (request == null)
         {
-            return ListUtils.EMPTY_LIST;
+            return Collections.EMPTY_LIST;
```
Comment (tdonohue) [inline dspace-api/src/main/java/org/dspace/authenticate/IPAuthentication.java]: In several places in the PR, it looks like you replace `ListUtils` with `Collections`, but leave the `import org.apache.commons.collections4.ListUtils` in place?  Is this on purpose, or should the `import` simply be removed here?

**Comment by Raf-atmire** [comment]
Incase the ListUtils is not used anywhere else in the class, it should
indeed be removed as Collections.EMPTY_LIST is not dependent on that
import. I will remove the unused imports as soon as possible.

**Comment by tdonohue** [comment]
NOTE: This PR looks to currently be failing approval from Snyk (which checks for possible security issues) because we still have a dependency on "commons-beanutils" v1.9.2 which pulls in "commons-collections" version 3.2.1 :(  Here's the report from Snyk

> Introduced through: DSpace/DSpace@DSpace/DSpace#d742d1a86f02eea69d6f982ddbf787f8046e8776 › commons-beanutils:commons-beanutils@1.9.2 › commons-collections:commons-collections@3.2.1

Looking at commons-beanutils, it looks like they have a ticket to update to commons-collections version 4.0 here: https://issues.apache.org/jira/browse/BEANUTILS-500   But, their version 2.0.0 isn't yet released.

So, while this PR is looking good, we won't completely get rid of commons-collections version 3 until we are able to update commons-beanutils.

**Comment by tomdesair** [comment]
Hi @tdonohue, `commons-beanutils` seems to be an optional dependency of Commons Configuration. Do you know if we actual need and use that library?

**Comment by tdonohue** [comment]
@tomdesair : Aha, glad you found the reason why it's in our dependencies. I couldn't recall, but we do use Configuration Builders to load our configs and provide the ability to "override" configs in dspace.cfg with those in local.cfg, etc.  Here's where they are used:
https://github.com/DSpace/DSpace/blob/master/dspace-services/src/main/java/org/dspace/servicemanager/config/DSpaceConfigurationService.java#L491

So, it's rather annoying here that we aren't going to be able to get rid of `commons-collections` version 3 until `commons-beanutils` is updated, and then `commons-configuration` updates to the latest `commons-beanutils`.  I don't see any easier way around this.

File: pom.xml [hunk 2]
```diff
@@ -1096,16 +1096,16 @@
...
-                <groupId>commons-collections</groupId>
-                <artifactId>commons-collections</artifactId>
-                <version>3.2.2</version>
-                <!-- <version>3.1</version> xmlui - wing -->
-            </dependency>
             <dependency>
                 <groupId>org.apache.commons</groupId>
                 <artifactId>commons-collections4</artifactId>
                 <version>4.1</version>
+                <!-- <version>3.1</version> xmlui - wing -->
```
Comment (mwoodiupui) [inline pom.xml]: We don't need this comment now, since XMLUI is no more.

**Comment by tdonohue** [comment]
Please rebase this PR against the latest `master` branch.  The `master` branch now requires adherence to our new Java Code Style documented at: https://wiki.duraspace.org/display/DSPACE/Code+Style+Guide

If you have any questions, feel free to ask in this PR or on our Slack: https://wiki.duraspace.org/display/DSPACE/Slack

**Comment by tdonohue** [comment]
@Raf-atmire : unfortunately, this PR seems to have similar issues to #1964 now (though not to the extreme of #1964) In aligning this PR with the new Code Style, the number of lines changed has now increased, as code you didn't change is being re-aligned / re-wrapped by your IDE.  I'm guessing you may have done some sort of bulk IDE rewrapping that has caused both of these PRs to now try and update the wrapping of unrelated code?

If you are bulk-rewrapping code in IntelliJ, I'd recommend using the same settings that were used to align everything with the Code Style. Otherwise, your IDE will likely continue to make unnecessarily realignment / rewrapping changes that will clutter your PRs. For these IntelliJ settings see: https://wiki.duraspace.org/display/DSPACE/Code+Style+Guide#CodeStyleGuide-ConfigureIDEAforbulkcleanup

The other option is to simply install the Checkstyle plugin for your IDE, in which case it will highlight code style issues *as you write your code*, allowing you to manually fix them as you go, instead of relying on bulk tools to properly format your code.  For the proper checkstyle plugin see: https://wiki.duraspace.org/display/DSPACE/Code+Style+Guide#CodeStyleGuide-IDESupport

**Comment by tomdesair** [comment]
@tdonohue I've restored the lines which should not have been modified. I've also rebased on master.

**Comment by tdonohue** [comment]
@tomdesair : thanks! Looks like there's a minor compilation failure though from Travis: https://travis-ci.org/DSpace/DSpace/builds/405388711#L3363 

In meantime, I'll start to give this an updated review

File: pom.xml [hunk 3]
```diff
@@ -1149,17 +1149,16 @@
...
-            </dependency>
             <dependency>
                 <groupId>org.apache.commons</groupId>
                 <artifactId>commons-collections4</artifactId>
                 <version>4.1</version>
             </dependency>
+            <dependency>
+                <groupId>commons-collections</groupId>
+                <artifactId>commons-collections</artifactId>
+                <version>3.2.2</version>
```
Comment (tdonohue) [inline pom.xml]: This dependency should be dropped, correct?

File: dspace-api/src/main/java/org/dspace/eperson/dao/impl/EPersonDAOImpl.java [hunk 4]
```diff
@@ -96,14 +95,14 @@ public int searchResultCount(Context context, String query, List<MetadataField>
                                                       .toLowerCase() + " FROM EPerson as " + EPerson.class
             .getSimpleName().toLowerCase();
 
-        List<MetadataField> sortFields = ListUtils.EMPTY_LIST;
+        List<MetadataField> sortFields = Collections.EMPTY_LIST;
 
-        if (metadataSortField != null) {
+        if (metadataSortField != null) {ItemRestRepositoryIT
```
Comment (tdonohue) [inline dspace-api/src/main/java/org/dspace/eperson/dao/impl/EPersonDAOImpl.java]: Here's the compilation failure.  A random `ItemRestRepositoryIT` was accidentally included here.

File: pom.xml (hunk 3 above)
Comment (tomdesair) [reply pom.xml]: No, we still have the issue with `beanutils` mentioned above. Removing that dependency results in:

```
[WARNING] 
Dependency convergence error for commons-collections:commons-collections:3.2.1 paths to dependency are:
+-org.dspace:dspace-api:7.0-SNAPSHOT
  +-commons-validator:commons-validator:1.5.0
    +-commons-beanutils:commons-beanutils:1.9.2
      +-commons-collections:commons-collections:3.2.1
and
+-org.dspace:dspace-api:7.0-SNAPSHOT
  +-commons-validator:commons-validator:1.5.0
    +-commons-collections:commons-collections:3.2.2

[WARNING] Rule 0: org.apache.maven.plugins.enforcer.DependencyConvergence failed with message:
Failed while enforcing releasability the error(s) are [
Dependency convergence error for commons-collections:commons-collections:3.2.1 paths to dependency are:
+-org.dspace:dspace-api:7.0-SNAPSHOT
  +-commons-validator:commons-validator:1.5.0
    +-commons-beanutils:commons-beanutils:1.9.2
      +-commons-collections:commons-collections:3.2.1
and
+-org.dspace:dspace-api:7.0-SNAPSHOT
  +-commons-validator:commons-validator:1.5.0
    +-commons-collections:commons-collections:3.2.2
]
```

Comment (tdonohue) [reply pom.xml]: @tomdesair : Aha, yes, I neglected to reread the discussion above.  You are correct, we do need it still.

**Review by tdonohue — APPROVED** [review]
👍 Code looks correct, and it passes all tests/build.

I am going to add only minor addition to this PR -- a comment to the main POM about why we still need `commons-collections` v3.  However, once that is applied, I think this can be merged.

**Pull request merged**